datalad/distribution/publish.py

```diff
@@ -219,14 +219,29 @@ def __call__(dataset=None, to=None, path=None, recursive=False,
 
             lgr.info("Publishing dataset {0} to sibling {1} "
                      "...".format(ds, dest_resolved))
+
             # we now know where to push to:
-            ds.repo.push(remote=dest_resolved,
-                         refspec=ds.repo.get_active_branch(),
-                         set_upstream=set_upstream)
+            push_result = ds.repo.push(remote=dest_resolved,
+                                       refspec=ds.repo.get_active_branch(),
+                                       set_upstream=set_upstream)
+
+            from git.remote import PushInfo as PI
+            for push_info in push_result:
+                if push_info.flags & PI.ERROR == PI.ERROR:
```

Comment:
even though there is probably clear priority of having binary operands applied first, I would appreciate having parenthesis around the binary operation before giving it to comparison ;)